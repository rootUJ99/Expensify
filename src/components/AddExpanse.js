import React from 'react';
import ReactDOM from 'react-dom';
const AddExpanse = () => (
    <div>
        this is add page
    </div>
);
export default AddExpanse;