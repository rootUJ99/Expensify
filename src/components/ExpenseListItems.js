import React from 'react';
import {Link} from 'react-router-dom';
const ExpenseListItem = ({ id, dispatch, description, amount, createdAt})=>(
    <div>
    <h3><Link to={`/edit/${id}`}>{description}</Link></h3>
    <h4>{amount}-{createdAt}</h4>
       
    </div>
);

export default ExpenseListItem;
