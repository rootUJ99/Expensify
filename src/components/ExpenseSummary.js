import React from 'react';
import ReactDOM from 'react-dom';
import { connect } from 'react-redux';
import numeral from 'numeral';
import selectExpenses from '../selectors/expenses';
import selectExpensesTotal from '../selectors/expenses-total';
export const ExpenseSummary = ({ expenseCount, expensesTotal }) => {
    const expenseWord = expenseCount === 1 ? 'expense' : 'expenses';
    const formattedExpensesTotal = numeral(expensesTotal).format('$0,0.00');
    return (
        <div>
            <h1>Viewing {expenseCount} {expenseWord} totalling {formattedExpensesTotal}</h1>
        </div>
    );
}
const mapStateToProps=(state)=>{
    const visiblityExpenses = selectExpenses(state.expenses,state.filters);
    return{
        expenseCount:visiblityExpenses.length,
        expensesTotal:selectExpensesTotal(visiblityExpenses)
    };
};
export default connect(mapStateToProps)(ExpenseSummary);
