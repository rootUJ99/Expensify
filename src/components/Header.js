import React from 'react';
import ReactDOM from 'react-dom';
import { BrowserRouter, Route, Switch, Link, NavLink } from 'react-router-dom';
const Header = () => (
    <header>
        <h1>Expensify</h1>
        <NavLink to="/" activeClassName="active-nav" exact={true}>Dashboard</NavLink>
        <NavLink to="/create" activeClassName="active-nav">Create</NavLink>
        <NavLink to="/edit" activeClassName="active-nav">Edit</NavLink>
    </header>
);
export default Header;