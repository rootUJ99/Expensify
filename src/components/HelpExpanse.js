import React from 'react';
import ReactDOM from 'react-dom';
const HelpExpanse = () => (
    <div>
        this is help
    </div>
);
export default HelpExpanse;