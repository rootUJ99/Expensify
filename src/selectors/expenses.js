import moment from 'moment';
//Get visibility Expenses
export default (expenses, { text, sortBy, startDate, endDate }) => {
    return expenses.filter((expense) => {
        //for 
        const createdAtMoment=moment(expense.createdAt);
        //for calander
        const startDateMatch =startDate ? startDate.isSameOrBefore(createdAtMoment, 'day') :true
        //for calander
        const endDateMatch = endDate ? endDate.isSameOrAfter(createdAtMoment, 'day') : true
        //for searching purpose
        const textMatch = typeof text !== 'string' || (expense.description.toLocaleLowerCase()).includes(text.toLocaleLowerCase())
        return startDateMatch && endDateMatch && textMatch
        //for select menu
    }).sort((a, b) => {
        if (sortBy === 'date') {
            return a.createdAt < b.createdAt ? 1 : -1;
        }
        else if (sortBy === 'amount') {
            return a.amount < b.amount ? 1 : -1;
        }
    })
}
