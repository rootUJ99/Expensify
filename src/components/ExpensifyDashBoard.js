import React from 'react';
import ReactDOM from 'react-dom';
import ExpenseList from './ExpenseList';
import ExpenseListItems from './ExpenseListItems';
import ExpenseListFilters from './ExpenseListFilters';
const ExpensifyDashBoard = () => (
    <div>
        <ExpenseListFilters/>
        <ExpenseList/>
    </div>
);
export default ExpensifyDashBoard;