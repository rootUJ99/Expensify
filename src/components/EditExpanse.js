import React from 'react';
import { connect } from 'react-redux';
import ExpenseForm from './ExpenseForm';
import { startEditExpense, startRemoveExpense } from '../actions/expenses';
export class EditExpanse extends React.Component {
    onSubmit = (expense) => {
        this.props.startEditExpense(this.props.expense.id, expense);
        this.props.history.push('/');
        console.log('updated', expense);
    }
    onRemove = () => {
        this.props.startRemoveExpense({ id: this.props.expense.id })
        this.props.history.push('/');
    }
    render() {
        return (
            <div>
                <div className="page-header" >
                    <div className="content-container">
                        <h1 className="page-header__title">Edit Expense</h1>
                    </div>
                </div>
                <div className="content-container">
                <ExpenseForm
                    expense={this.props.expense}
                    onSubmit={this.onSubmit}
                />
                <button
                    className="button2 button2--remove"
                    onClick={this.onRemove}
                > Remove Expense</button>
            </div>
            </div>
        );
    }
}

const mapStateToProps = (state, props) => {
    return {
        expense: state.expenses.find((expense) => expense.id === props.match.params.id)
    };
};
const mapDispathToProps = (dispatch, props) => ({
    startEditExpense: (id, expense) => dispatch(startEditExpense(id, expense)),
    startRemoveExpense: (data) => dispatch(startRemoveExpense(data))
});
export default connect(mapStateToProps, mapDispathToProps)(EditExpanse);