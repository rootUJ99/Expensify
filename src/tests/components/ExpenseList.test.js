import React from 'react';
import { shallow } from 'enzyme';//read about shallow later on
import { ExpenseList } from '../../components/ExpenseList';
import expenses from '../fixtures/expenses';
//render expenseList 
test('sohuld render ExpenseList with expenses',()=>{
    const wrapper= shallow(<ExpenseList expenses={expenses} />);
    expect(wrapper).toMatchSnapshot();
});
//render empty expenseList
test('should render ExpenseList with empty message',()=>{
    const wrapper = shallow(<ExpenseList expenses={[]} />);
    expect(wrapper).toMatchSnapshot();
});