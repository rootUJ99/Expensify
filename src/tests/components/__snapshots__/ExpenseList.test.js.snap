// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render ExpenseList with empty message 1`] = `
<div>
  <p>
    No Expenses
  </p>
</div>
`;

exports[`sohuld render ExpenseList with expenses 1`] = `
<div>
  <ExpenseListItem
    amount={195}
    createdAt={0}
    description="Gum"
    id={1}
    key="1"
    note=""
  />
  <ExpenseListItem
    amount={200156}
    createdAt={-345600000}
    description="Rent"
    id={2}
    key="2"
    note=""
  />
  <ExpenseListItem
    amount={250}
    createdAt={345600000}
    description="Coffee"
    id={3}
    key="3"
    note=""
  />
</div>
`;
